lld/ELF/ScriptLexer.cpp

```diff
@@ -231,6 +231,23 @@ ScriptLexer::Token ScriptLexer::till(StringRef tok) {
   return {};
 }
 
+ScriptLexer::Token ScriptLexer::readNameTill(StringRef tok) {
+  // this behaves like till but expects that token to be an
```

Comment:
Perhaps assert that `tok[0] != '"'`. 

If `next()` is quoted, we should return it early, as `"("` should be allowed if `tok == "("`.